docbook/reference/en/en-US/modules/Reactive.xml

```diff
@@ -1,63 +1,810 @@
+       through which objects pass.
+    </para>
+    
+    <para>
+       One important aspect of the specification is flow control, the ability of a <classname>Suscriber</classname>
+       to control the load it receives from a <classname>Producer</classname> by calling
+       <methodname>Suscription.request()</methodname>. The general term in this context for flow control is
+       <emphasis role="bold">backpressure</emphasis>.
+    </para>
+    
+    <para>
+       There are a number of implementations of Reactive Streams, including 
+    </para>
+    
+    <orderedlist>
+       <listitem><emphasis role="bold">RxJava</emphasis>: 
+          <ulink url="https://github.com/ReactiveX/RxJava">https://github.com/ReactiveX/RxJava</ulink> (end of life, superceded by RxJava 2)
+       </listitem>
+       <listitem><emphasis role="bold">RxJava 2</emphasis>: 
+          <ulink url="https://github.com/ReactiveX/RxJava">https://github.com/ReactiveX/RxJava</ulink>
+       </listitem>
+       <listitem><emphasis role="bold">Reactor</emphasis>: 
+          <ulink url="http://projectreactor.io/">http://projectreactor.io/</ulink>
+       </listitem>
+       <listitem><emphasis role="bold">Flow</emphasis>: 
+          <ulink url="https://community.oracle.com/docs/DOC-1006738">https://community.oracle.com/docs/DOC-1006738/</ulink>:
+          (Java JDK 9+)
+          </listitem>
+    </orderedlist>
+    
+    <para>
+       RESTEasy currently supports RxJava (deprecated) and RxJava2.
+    </para>
+    
+</sect1>
+
+<sect1>
+<title>Pluggable reactive types: RxJava 2 in RESTEasy</title>
+
+    <para>
+       JAX-RS 2.1 doesn't require support for any Reactive Streams implementations, but it does allow
+       for extensibility to support various reactive libraries.
+       RESTEasy's optional modules <code>resteasy-rxjava1</code> and <code>resteasy-rxjava2</code> 
+       add support for <link linkend="https://github.com/ReactiveX/RxJava">RxJava 1 and 2</link>.
+       [Only <code>resteasy-rxjava2</code> will be discussed here, since <code>resteasy-rxjava1</code> is
+       deprecated, but the treatment of the two are quite similar.]
+    </para>
+    
+    <para>
+       In particular, <code>resteasy-rxjava2</code>
+       contributes support for reactive types <classname>io.reactivex.Single</classname>,
+       <classname>io.reactivex.Flowable</classname>, and <classname>io.reactivex.Observable</classname>.
+       Of these, <classname>Single</classname> is similar to <classname>CompletionStage</classname> in that
+       it holds at most one potential value. <classname>Flowable</classname> implements 
+       <classname>io.reactivex.Publisher</classname>, and <classname>Observable</classname> is very
+       similar to <classname>Flowable</classname> except that it doesn't support backpressure.
+       So, if you import <code>resteasy-rxjava2</code>, you can just start returning these reactive types from your

```

Comment:
@ronsigal, @asoldano: "import resteasy-rxjava2" ... Are there some future plans to add resteasy-rxjava2 module to wf? (I know, that this can't be in scope of this PR and  in scope of RESTEASY-1798)

Currently, if I want to import resteasy-rxjava2 to WF, I need to copy ${RESTEASY_GIT}/jboss-modules/target/modules/org/jboss/resteasy/resteasy-rxjava2 directory to WF and add resteasy-rxjava2 dependency to jboss-deployment-structure.xml file in deployment ... and it is quite complicated in my PoV ...

Reply:
My opinion here is that we'll have to add the resteasy-rxjava and resteasy-rxjava2 modules into wildfly, possibly as tech-preview

Reply:
Can we add resteasy-rxjava2 only? (without resteasy-rxjava(1))

Reply:
Hi @marekkopecky,

1. Re the Rx2FlowableTest error messages: That sounds familiar. I added 
>&lt;module name="javax.ws.rs.api"/>

to the resteasy-rxjava2 module.xml file in jboss-modules a while ago (at home). I think that's what fixed it. I should have a new commit soon.

2. Re jboss-deployment-structure.xml: Is that necessary? I just unzip resteasy-jboss-modules-4.0.0-SNAPSHOT.zip

3. Re wildfly: I defer to Alessio on that question.

4. Re Just resteasy-rxjava2: I guess the question is: How far along is the rx community in upgrading from rxjava to rxjava2. @marekkopecky, @FroMage, what do you think?

Reply:
Hi @ronsigal 

 2. You are right, resteasy-jboss-modules-4.0.0-SNAPSHOT.zip is much more easier, thank you!

 4. In my PoV, we shouldn't add to WF something, that is at the end of life (= rxjava(1)). If there occurs some bug in rxjava, I assume that this bug won't be fixed (cc: @asoldano )